internal/app/orchestrator/orchestrator.go

```diff
@@ -0,0 +1,31 @@
+package orchestrator
+
+import (
+	gcp "github.com/envoyproxy/go-control-plane/pkg/cache"
+)
+
+// Orchestrator provides an interface that handles the discovery requests from sidecars.
+type Orchestrator interface {
+	// CreateWatch returns a new open watch from a non-empty request.
+	// Returns a channel with Response and a function which is a callback for handling stream cancellations.
+	CreateWatch(req gcp.Request) (chan gcp.Response, func())
+}
+
+// orchestrator provides a mechanism to implement the Orchestrator interface.
+// The orchestrator maintains all the relevant caches and makes sure of the following:
```

Comment:
I'd put this comment over the public `Orchestrator` interface or as a package level comment imo.